From 6eeaa6d172999149fb26305b664cc38c5d2e8b7c Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <ross.lagerwall@citrix.com>
Date: Wed, 27 May 2015 10:04:00 +0100
Subject: [PATCH] libxl: Don't insert PCI device into xenstore for HVM guests

When doing passthrough of a PCI device for an HVM guest, don't insert
the device into xenstore, otherwise pciback attempts to use it which
conflicts with QEMU.

Signed-off-by: Ross Lagerwall <ross.lagerwall@citrix.com>
diff --git a/tools/libxl/libxl_pci.c b/tools/libxl/libxl_pci.c
index b5444d1552..cd2f84981c 100644
--- a/tools/libxl/libxl_pci.c
+++ b/tools/libxl/libxl_pci.c
@@ -1297,6 +1297,7 @@ static void pci_add_dm_done(libxl__egc *egc,
     STATE_AO_GC(pas->aodev->ao);
     libxl_ctx *ctx = libxl__gc_owner(gc);
     libxl_domid domid = pas->pci_domid;
+    libxl_domain_type type = libxl__domain_type(gc, domid);
     char *sysfs_path;
     FILE *f;
     unsigned long long start, end, flags, size;
@@ -1414,7 +1415,8 @@ static void pci_add_dm_done(libxl__egc *egc,
         }
     }
 
-    if (!starting && !libxl_get_stubdom_id(CTX, domid))
+    if (!starting && !libxl_get_stubdom_id(CTX, domid) &&
+        type == LIBXL_DOMAIN_TYPE_PV)
         rc = libxl__device_pci_add_xenstore(gc, domid, pcidev, starting);
     else
         rc = 0;
